sdk/src/main/kotlin/io/rover/sdk/ui/blocks/poll/VotingInteractor.kt

```diff
@@ -0,0 +1,28 @@
+package io.rover.sdk.ui.blocks.poll
+
+import io.rover.sdk.data.graphql.ApiResult
+import io.rover.sdk.streams.PublishSubject
+import io.rover.sdk.streams.subscribe
+
+internal class VotingInteractor(
+    private val votingService: VotingService,
+    private val votingStorage: VotingStorage,
+    val optionResults: PublishSubject<OptionResults> = PublishSubject()) {
```

Comment:
Why can the internal subject be replaced by something externally constructing the VotingInteractor?  This would seem to allow you to override the Interactor's functionality in unexpected ways.

Reply:
I did this in order to mock the subject so that I can verify onNext is called on it. I agree that it's not ideal and that we shouldn't be modifying code in ways that don't make much sense for tests

Reply:
consider using .block(timeout=).first() to wait for something to be emitted